apps/portals-base/src/App.tsx

```diff
@@ -1,28 +1,32 @@
 import { BrowserRouter, Switch } from 'react-router-dom'
-import React from 'react'
-import './App.scss'
-import Footer from './Footer'
-import AppInitializer from './AppInitializer'
-import Navbar from './Navbar'
+import React, { useMemo } from 'react'
+import Footer from './components/Footer'
+import AppInitializer from './components/AppInitializer'
+import Navbar from './components/Navbar'
 import { CookiesProvider } from 'react-cookie'
 import {
   CookiesNotification,
   SynapseTheme,
   SynapseToastContainer,
   defaultQueryClientConfig,
 } from 'synapse-react-client'
-import { LogInDialogContextProvider } from './LogInDialogContext'
+import { LogInDialogContextProvider } from './components/LogInDialogContext'
 import { createTheme, ThemeProvider } from '@mui/material'
-import palette from './config/paletteConfig'
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { PortalProps } from './components/PortalProps'
+import { PortalContextProvider } from './components/PortalContext'
 
-const RouteResolver = React.lazy(() => import('./RouteResolver'))
-const theme = createTheme(SynapseTheme.mergeTheme({ palette }))
+const RouteResolver = React.lazy(() => import('./components/RouteResolver'))
 const queryClient = new QueryClient(defaultQueryClientConfig)
 
-function App() {
+function App(props: PortalProps) {
```

Comment:
Each portal can render this  `App` component but they must pass `PortalProps`, which basically encapsulates a portal configuration.